x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/predicate/Range.java

```diff
@@ -88,21 +89,25 @@ public ZoneId zoneId() {
     @Override
     public boolean foldable() {
         if (lower.foldable() && upper.foldable()) {
-            return areBoundariesInvalid() || value.foldable();
+            // TODO rework this to not use unbounded
+            Object lowerValue = lower.fold(FoldContext.unbounded());
+            Object upperValue = lower.fold(FoldContext.unbounded());
```

Comment:
This is wrong. Copy-pasta mistake probably.

Reply:
Fun! It's interesting that tests didn't catch this. Let me make a test that fails....

Reply:
Indeed. There are many edge cases, we can create tests for as many as we can think of. There will always be something left behind, I came to believe it’s impossible to cover everything.

Reply:
Er, alex's test catches it already.

Reply:
Yeah. I'm not accusing at all. We try to cover it all. I try not to make mistakes too. But I figure if I've made a mistake once it's good to have a case.

Reply:
To be fair, I think we just never fold ranges in actual queries, at all; so it makes sense that no test caught this in the past. It  doesn't help that `Range` was inherited from `ql`.